lib/bolt/plugin/module.rb

```diff
@@ -170,7 +170,10 @@ def process_params(task, opts)
           params = @config ? config.merge(params) : params
           validate_params(task, params)
         end
-        params['_boltdir'] = @context.boltdir.to_s
+
+        if local
+          params['_boltdir'] = @context.boltdir.to_s
```

Comment:
Logically it seems like this should be set on `meta` rather than `params`. And that alone should be enough to fix the bug, since the `puppet_library` method already ignores the metaparams (except for `_run_as` which it pulls out explicitly).

Reply:
good catch.